transport/src/androidTest/kotlin/com/genesys/cloud/messenger/transport/core/StateMachineTest.kt

```diff
@@ -204,6 +204,30 @@ class StateMachineTest {
         verify { mockStateChangedListener(expectedStateChange) }
     }
 
+    @Test
+    fun verifyFullStateTransitionCycle() {
+        subject.onConnect()
+        assertThat(subject).isConnecting()
+        subject.onConnectionOpened()
+        assertThat(subject).isConnected()
+        subject.onConfiguring()
+        subject.onSessionConfigured(connected = true, newSession = true)
+        assertThat(subject).isConfigured(connected = true, newSession = true)
+        subject.onReconnect()
+        assertThat(subject).isReconnecting()
+        subject.onConnect()
+        assertThat(subject).isReconnecting()
+        subject.onConnectionOpened()
+        assertThat(subject).isReconnecting()
+        subject.onConfiguring()
+        subject.onSessionConfigured(connected = true, newSession = true)
```

Comment:
in most cases after reconnecting, should it be `newSession = false`?

Reply:
It should, but since it is something we receive from BE it does not really important to capture it here. All we want to do is to validate that parameters are not modified and that they match the one that are passed in. 
I will update the `newSession` to `false` to keep the flow clean.